ext/vscode/src/commands/installCli.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { DialogResponses, IActionContext } from '@microsoft/vscode-azext-utils';
+import * as os from 'os';
+import { localize } from '../localize';
+import { resetAzdInstalledCheck } from '../utils/azureDevCli';
+import { executeInTerminal } from '../utils/executeInTerminal';
+import { getAzDevTerminalTitle } from './cmdUtil';
+
+const WindowsTerminalCommand = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"`;
+const LinuxTerminalCommand = `curl -fsSL https://aka.ms/install-azd.sh | bash`;
+const MacTerminalCommand = LinuxTerminalCommand; // Same as Linux
+
+export async function installCli(context: IActionContext, shouldPrompt: boolean = true): Promise<void> {
+    if (shouldPrompt) {
+        const message = localize('azure-dev.commands.cli.install.prompt', 'This will install or update the Azure Developer CLI. Do you want to continue?');
+        // Don't need to check the result, if the user chooses cancel a UserCancelledError will be thrown
+        await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, DialogResponses.cancel);
+    }
+
+    let terminalCommand: string;
+
+    if (os.platform() === 'win32') {

```

Comment:
I'm guessing `win32` is for both x86 and x64?

Reply:
Yes. Although, I should use the `isWindows()` utility we have in the utils package.